src/queue.jl

```diff
@@ -6,31 +6,57 @@ Each queue is uniquely associated with an agent.
 """
 mutable struct HSAQueue
     agent::HSAAgent
-    queue::Ref{Ptr{HSA.Queue}}
+    queue::Ptr{HSA.Queue}
     active::Bool
 end
-
+const QUEUES = WeakKeyDict{Ptr{HSA.Queue},HSAQueue}()
 const DEFAULT_QUEUES = IdDict{HSAAgent,HSAQueue}()
 
+function queue_error_handler(status::HSA.Status, _queue::Ptr{HSA.Queue}, data::Ptr{Cvoid})::Nothing
+    # TODO: From which thread are we calling this?
```

Comment:
The HSA RT docs say, "it will be invoked in an HSA runtime thread", so we should assume that we need to be careful about accessing shared globals.